kubernetes/config-maps/nginx-proxy/configs/http.conf.j2

```diff
@@ -5,12 +5,90 @@ map $http_accept $extension {
     application/json json;
 }
 
+{% if concent_versions[concent_version].load_balancer_turn_on %}
+upstream {{ concent_versions[concent_version].latest_golem_message_version }} {
+     server nginx-proxy.default.svc.cluster.local;
+}
+
+upstream {{ concent_versions[concent_version].average_golem_message_version }} {
+     {% set concent_old_version = concent_versions.keys() |list %}
+     server {{ concent_versions[concent_old_version[1]].cluster_address }};
+}
+
+{% set latest_golem_message_version = concent_versions[concent_version].latest_golem_message_version %}
+{% set average_golem_message_version = concent_versions[concent_version].average_golem_message_version %}
+# Map to different upstream backends based on the `X-Golem-Messages` http header
+map $http_x_golem_messages $version {
+
+     ""          "http://{{ latest_golem_message_version }}";
+     {{ latest_golem_message_version }}   "http://{{ latest_golem_message_version }}";
+     {{ average_golem_message_version }}  "http://{{ average_golem_message_version }}";
+     default    "not_supported";
+     # Check semi version 2.0 format
+     ~^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$ 
+     "not_found";
+}
+
+map $http_x_golem_messages $host_param {
+     ""                                    internal.local;
+     {{ latest_golem_message_version }}    internal.local;
+     {{ average_golem_message_version }}   $http_host;
+     default                               $http_host;
+     # Check semi version 2.0 format
+     ~^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$ 
```

Comment:
Done

Reply:
That's not really what I had in mind. I meant a nginx variable.

Actually, where is this this regex defined now? I no longer see it in the pull request...

Reply:
The nginx variable can only set in `context: server, location, if`. According to http://nginx.org/en/docs/http/ngx_http_rewrite_module.html#set. In this place are not require because default support all example so `$http_host`. We only change this value in current golem message version.

Reply:
The regex is only use in switching between upstream server.

Reply:
So it's in the `var.yml` file now? That's not good. This file is meant for configuration only.

If it's not possible with a nginx variable let's just leave it as it was before.